code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -670,10 +670,14 @@
 						return
 					if(isnull(personnel_list[input]))
 						return
-					var/icon/character_icon = personnel_list[input]
+					var/character_icon = personnel_list[input]
 					if(character_icon)
 						qdel(holo_icon)//Clear old icon so we're not storing it in memory.
-						holo_icon = getHologramIcon(icon(character_icon))
+						var/static/atom/movable/screen/appearance_holder = new()
+						appearance_holder.appearance = character_icon
+
+						var/icon/icon_for_holo = getFlatIcon(appearance_holder)
```

Comment:
I think you could actually just pass the appearance into getFlat, since it's a subtype off /image, which should be supported just fine